core/scrollbar.js

```diff
@@ -190,22 +188,36 @@ Blockly.Scrollbar = function(workspace, horizontal, opt_pair) {
   this.workspace_ = workspace;
   this.pair_ = opt_pair || false;
   this.horizontal_ = horizontal;
-  this.oldHostMetrics_ = {};
+  this.oldHostMetrics_ = null;
 
   this.createDom_();
 
+  this.barLength = 0;
```

Comment:
Good point.  Originally they weren't because ScrollbarPair was changing them directly, but now I think it's only changing them through the set* calls.  Except xCoordinate and yCoordinate, so I'll leave those public for now.